### ISSUE TITLE
All channels have to be closed by now.

### ISSUE THREAD

**Issue by ddzobov** [jakubkulhan/bunny#105]
Hello
Sometimes i’m getting LogicException “ All channels have to be closed by now.” in test environment.

Can some one explain me, why i can get this error?

**Comment by WyriHaximus** [comment]
That error occurs when there are still open channels when performing a disconnect, something that shouldn't happen

**Comment by ddzobov** [comment]
How to prevent getting this error?

**Comment by WyriHaximus** [comment]
So to get a more complete picture: Are you able to connect and produce or consume messages?

Because if you're not, that is probably the root cause of your issue.

**Comment by ddzobov** [comment]
Yes, app with this lib working together with swoole (reusing rabbitmq connect between requests).
In production i have no problems because there is many in-out messages.
I’m catching this error at test environment by running daily tests, so this error occurs after some idle about 3-4h.

**Comment by WyriHaximus** [comment]
Sooo do you share swooles event loop (through an adapter) with bunny? If not, that's highly likely the reason, as the existing connection will simply timeout and close.
